index.js

```diff
@@ -1,117 +0,0 @@
-  treeForVendor: function (tree) {
-    this._findPretenderPaths();
-
-    var pretenderTree = new Funnel(this._pretenderDir, {
-      files: [path.basename(this._pretenderPath)],
-      destDir: '/pretender',
-    });
-
-    var routeRecognizerFilename = path.basename(this._routeRecognizerPath);
-    var routeRecognizerTree = new Funnel(
-      path.dirname(this._routeRecognizerPath),
-      {
-        files: [routeRecognizerFilename, routeRecognizerFilename + '.map'],
-        destDir: '/route-recognizer',
-      }
-    );
-
-    var fakeRequestTree = new Funnel(path.dirname(this._fakeRequestPath), {
-      files: [path.basename(this._fakeRequestPath)],
-      destDir: '/fake-xml-http-request',
-    });
-
-    var trees = [
-      tree,
-      pretenderTree,
-      routeRecognizerTree,
-      fakeRequestTree,
-      // tree is not always defined, so filter out if empty
-    ].filter(Boolean);
-
-    return new MergeTrees(trees, {
-      annotation: 'pretender-and-friends: treeForVendor',
-    });
-  },
-
-  included: function (app) {
-    this._super.included.apply(this, arguments);
-
-    this.setupFactoryGuyInclude(app);
-
-    if (this.includeFactoryGuyFiles) {
-      this._findPretenderPaths();
-
-      app.import(
-        'vendor/fake-xml-http-request/' + path.basename(this._fakeRequestPath)
-      );
-      app.import(
-        'vendor/route-recognizer/' + path.basename(this._routeRecognizerPath)
-      );
-      app.import('vendor/pretender/' + path.basename(this._pretenderPath));
-
-      // this seems like a stupid thing to do, but it is needed in fastboot environment / borrowed it from mirage
-      // eventually what I should do is not load any factory guy files in fastboot environment,
-      // but that is a real pain, so for now this will do.
-      app.import('vendor/pretender-shim.js', {
-        type: 'vendor',
-        exports: { pretender: ['default'] },
-      });
-    }
-  },
-
-  setupFactoryGuyInclude: function (app) {
-    let defaultEnabled = /test|development/.test(app.env),
-      defaultSettings = { enabled: defaultEnabled },
-      userSettings = app.project.config(app.env).factoryGuy || {},
-      settings = Object.assign(defaultSettings, userSettings);
-
-    this.includeFactoryGuyFiles = settings.enabled;
-
-    // Have to be carefull not to exclude factory guy from addon tree
-    // in development or test env
-    let trees = /test|development/.test(app.env) ? 'app' : 'app|addon';
-    this.treeExcludeRegex = new RegExp(trees);
-  },
-
-  treeFor: function (name) {
-    // Not sure why this is necessary, but this stops the factory guy files
-    // from being added to app tree. Would have thought that this would have
-    // happened in treeForApp above, but not the case
-    if (
-      !this.includeFactoryGuyFiles &&
-      this.treeExcludeRegex &&
-      this.treeExcludeRegex.test(name)
-    ) {
-      return;
-    }
-    return this._super.treeFor.apply(this, arguments);
```

Comment:
TODO: implement this with embroider/macros

The rest of the pretender package shim stuff in this file shouldn't be necessary